File: Support/Fuego/Pythia/pythia-0.4/packages/journal/journal/Journal.py
#!/usr/bin/env python
# 
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 
#                               Michael A.G. Aivazis
#                        California Institute of Technology
#                        (C) 1998-2003 All Rights Reserved
# 
#  <LicenseText>
# 
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 


class Journal(object):


    def record(self, entry):
        self.device.record(entry)
        return


    def entry(self):
        from Entry import Entry
        return Entry()


    def channel(self, name, channel=None):
        if channel is None:
            return self._channels.get(name)

        self._channels[name] = channel

        return channel


    def channels(self):
        return self._channels.keys()


    def __init__(self, name, device=None):
        self.name = name
        self._channels = {}

        if device is None:
            from Console import Console
            device = Console()
        self.device = device

        return


# version
__id__ = "$Id$"

#  End of file 

File: Support/Fuego/Pythia/pythia-0.5/packages/journal/lib/Journal.cc
// -*- C++ -*-
//
//--------------------------------------------------------------------------------
//
//                              Michael A.G. Aivazis
//                       California Institute of Technology
//                       (C) 1998-2003  All Rights Reserved
//
// <LicenseText>
//
//--------------------------------------------------------------------------------
//

#include <portinfo>
#include <map>
#include <string>
#include <vector>
#include <ostream>

#include "Device.h"
#include "Journal.h"
#include "Index.h"
#include "Entry.h"

#include "StreamDevice.h"
#include "Console.h"


using namespace journal;

// meta-methods

Journal::Journal() :
    _device(new Console)
{}

Journal::~Journal() {
    delete _device;
}

// version
// $Id$

// End of file
